recipes/perl-forks/build.sh

```diff
@@ -1,2 +1,23 @@
 #!/bin/bash
-cpanm .
+
+# cpanm -i .
+
+# If it has Build.PL use that, otherwise use Makefile.PL
+if [ -f Build.PL ]; then
+    perl Build.PL
+    ./Build
+    ./Build test
+    # Make sure this goes in site
+    ./Build install --installdirs site
+elif [ -f Makefile.PL ]; then
+    # Make sure this goes in site
+    perl Makefile.PL # INSTALLDIRS=site
+    cat Makefile | grep fstack
+    sed -i 's/-fstack-protector-strong/-fstack-protector/g' Makefile
```

Comment:
The ExtUtils::MakeMaker adds `-fstack-protector-strong` _somewhere_, but I can't figure out where. That flag however is only valid in >= gcc 4.9. So, one hack is to switch to the older `-fstack-protector` which is acceptable, but I'm not really sure what the implications are for speed or security here. I also don't know why gcc-4.8 is being used, when `llvm` / `clang` should be used on the mac build, but this replacement does at least seem to work. Is it a valid approach?

Reply:
For now you should rather remove the `gcc` dependency as is done with `perl` and other Perl recipes. That way the "system" `gcc` from the Docker container is used. That one is also just `gcc 4.8.2` but a patched one, see: https://github.com/conda-forge/perl-feedstock/pull/12#issuecomment-282003891.
(Note that this will change in future, but currently this is needed due to compatibility/legacy reasons.)

Reply:
Thanks @mbargull, those discussions help. I think that will work for the linux build, but is the patched `gcc 4.8.2` also used for the macOS build? I'm checking now, but I don't believe that will work locally with `circleci build` and I don't think there's any reason to believe it would be different on the hosted CI, right?

Reply:
Also, does that mean this guideline is wrong? https://bioconda.github.io/guidelines.html#c-c

Reply:
macOS builds should use `clang` and `-fstack-protector-strong` shouldn't be added there, I think. Let's just see if the CI chokes on it. And the local `circleci build` should also use that patched `gcc 4.8.2` -- if not, let me know.
Regarding the guidelines, I think (not 100 % sure, though) that it's just the Perl packages that are an oddball here. We should adjust the guidelines anyway in near future when we switch to a newer `conda-build` version. So, for now, I'd take them as "subject to change".